### DISCUSSION TITLE
Reusing {{VALUE}} in multiple macro steps

### DISCUSSION THREAD

**Discussion by mehanoid** [chhoumann/quickadd#697] [Q&A] [answered]
If I use the `{{VALUE}}` tag in multiple steps of a macro, then at each step a prompt reappears asking for the value to be entered.
Is there any way to somehow reuse the entered value in multiple steps?
I found a [couple](https://github.com/chhoumann/quickadd/discussions/500) [similar questions](https://github.com/chhoumann/quickadd/discussions/515), but they are about the `{{VALUE:<variable name>}}`. I can use `{{VALUE:<variable name>}}`, then the value is reused. But in that case it won't capture the selected text in the editor.

**Comment by chhoumann** [comment, marked as answer]
Coming back to this one very late - sorry about that.

Two answers, depending on where the value comes from:

**Selected text:** plain `{{VALUE}}` uses the current editor selection instead of prompting. So if you run the macro with text selected, every step that contains `{{VALUE}}` gets the selection and you are never prompted at all.

**Typed input, prompted once:** `{{VALUE}}` honors the reserved variable name `value`. Put a user script as the first command of the macro:

```js
module.exports = async (params) => {
    const selection = params.app.workspace.activeEditor?.editor?.getSelection();
    params.variables["value"] =
        selection || (await params.quickAddApi.inputPrompt("Value"));
};
```

Every later step's plain `{{VALUE}}` then reuses that value without prompting - and this script still prefers the selection when there is one, which covers exactly the case you described where `{{VALUE:<name>}}` reuses fine but loses the selection.

(The per-step prompt you saw is intentional for a bare `{{VALUE}}` with nothing supplied: each choice also has to work standalone, so an unanswered `{{VALUE}}` asks again.)

The reserved `value` name is documented on the format syntax page: https://quickadd.obsidian.guide/docs/FormatSyntax
